source/gloperate/source/pipeline/Pipeline.cpp

```diff
@@ -40,6 +40,10 @@ Pipeline::Pipeline(Environment * environment, const std::string & className, con
 
 Pipeline::~Pipeline()
 {
+    while (!m_stages.empty())
```

Comment:
Can you integrate a comment stating that ```removeStage``` will change ```m_stages``` and thus, no (reverse) iterators can be used here?